### ISSUE TITLE
`and_modify` method should be part of the HashMap's Documentation.

### ISSUE THREAD

**Issue by KarlWithK** [rust-lang/rust#98122]
I understand that `and_modify` deals with the `Entry` enum; however, I believe the method would be advantageous to developers when dealing with HashMaps. 

A common pattern I see when using hashmaps is something like this:

```rust
strings.into_iter().for_each(|string| {
    let value = map.entry(string).or_insert(0);
    *value += 1;
});
```

However, with `and_modify` this turns into this:
```rust
strings.into_iter().for_each(|string| {
    map.entry(string)
        .and_modify(|value| *value += 1)
        .or_insert(1);
});
```

Which leans more into idiomatic rust. I just learned of `and_modify` today, and I believe more rust users would use it if they knew it existed. I think the problem is that `and_modify` is part of the `Entry` API.

**Comment by eggyal** [comment]
To be clear, are you suggesting that the documentation for [`HashMap::entry`](https://doc.rust-lang.org/std/collections/hash_map/struct.HashMap.html#method.entry) should include an example using `and_modify`?  That seems pretty reasonable.

@rustbot label +A-docs +E-easy

**Comment by KarlWithK** [comment]
Yes, I am. Can I work on the PR for the issue? Thanks!

**Comment by SkiFire13** [comment]
> ```
> strings.into_iter().for_each(|string| {
>     let value = map.entry(string).or_insert(0);
>     *value += 1;
> });
> ```

I would shorten this to:
```
strings.into_iter().for_each(|string| {
    *map.entry(string).or_insert(0) += 1;
});
```

IMO this is more idiomatic than the `and_modify` example, but this is probably subjective. It just seems more linear and thus simplier to me:
- if an entry is empty the default value is 0 (this is in addition to what normally happens)
- then increase it (this always happens)

In contrast with `and_modify`:
- if there's an entry increase it (which is conditional on not being the first element)
- else set it to 1 (which is condition to being the first element)

Anyway with #98125 I suppose this could be closed?

**Comment by KarlWithK** [comment]
So honestly, I agree with what you are saying! However, if we look at the HashMap documentation right now we see this:

```rust
// update a key, guarding against the key possibly not being set
let stat = player_stats.entry("attack").or_insert(100);
*stat += random_stat_buff();
```

I believe that `and_modfiy` should be used for this type of example. However, for the example I gave which was not the best I admit, I agree that it would be simpler than to do it in one liner.
